golem-cli/tests/api_deployment_fileserver.rs

```diff
@@ -0,0 +1,324 @@
+        serde_json::to_writer(&mut writer, &api_definition)?;
+        writer.flush()?;
+    }
+
+    let definition: HttpApiDefinitionWithTypeInfo =
+        cli.run(&["api-definition", "add", api_path.to_str().unwrap()])?;
+
+    drop(temp_dir);
+
+    let _: ApiDeployment = cli.run(&[
+        "api-deployment",
+        "deploy",
+        &cfg.arg('d', "definition"),
+        &format!("{}/{}", definition.id, definition.version),
+        &cfg.arg('H', "host"),
+        &host,
+    ])?;
+
+    let res1 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/foo.txt"))?;
+    assert_eq!(res1.status().as_u16(), 200);
+    assert_eq!(res1.headers().get("content-type").unwrap(), "text/plain");
+    assert_eq!(res1.text()?, "foo\n");
+
+    let res2 = reqwest::blocking::get(format!("http://{host}/files/{component_id}/bar.txt"))?;
+    assert_eq!(res2.status().as_u16(), 200);
+    assert_eq!(res2.headers().get("content-type").unwrap(), "text/plain");
+    assert_eq!(res2.text()?, "bar\n");
+
+    Ok(())
+}
+
+fn api_deployment_fileserver_complex(
+    (deps, name, cli): (
+        &(impl TestDependencies + Send + Sync + 'static),
+        String,
+        CliLive,
+    ),
+) -> anyhow::Result<()> {
+    // the port is hardcoded in the env based test dependencies. Do not change this.
+    let host = "localhost:9093";
+    let component_name = format!("api_deployment_fileserver_complex{name}");
+    let golem_yaml = deps.component_directory().join("file-server/golem.yaml");
+
+    let cfg = &cli.config;
+    let component: ComponentView = cli.run_trimmed(&[
+        "component",
+        "add-with-manifest",
+        &cfg.arg('a', "app"),
+        golem_yaml.to_str().unwrap(),
+        &cfg.arg('c', "component-name"),
+        &component_name,
+    ])?;
+    let component_id = component.component_urn.id;
+
+    let temp_dir = tempfile::tempdir()?;
+
+    let api_definition = make_file_server_api_definition_complex(&component_id)?;
+    let api_path = temp_dir.path().join("api-file-server.json");
+
+    {
+        let mut api_file = std::fs::File::create_new(&api_path)?;
+        let mut writer = std::io::BufWriter::new(&mut api_file);
+        serde_json::to_writer(&mut writer, &api_definition)?;
+        writer.flush()?;
+    }
+
+    let definition: HttpApiDefinitionWithTypeInfo =
+        cli.run(&["api-definition", "add", api_path.to_str().unwrap()])?;
+
+    drop(temp_dir);
+
+    let _: ApiDeployment = cli.run(&[
+        "api-deployment",
+        "deploy",
+        &cfg.arg('d', "definition"),
+        &format!("{}/{}", definition.id, definition.version),
+        &cfg.arg('H', "host"),
+        &host,
+    ])?;
+
+    let res = reqwest::blocking::get(format!("http://{host}/files/{component_id}/file"))?;
+    assert_eq!(res.status().as_u16(), 201);
+    assert_eq!(res.headers().get("content-type").unwrap(), "application/json");
+    assert_eq!(res.text()?, "foo\n");
+
+    Ok(())
+}
+
+fn api_deployment_fileserver_stateful_worker(
```

Comment:
this guy is probably the single most interesting test in the pr as it touches almost everything I added.